docs/how-testing-works.md

```diff
@@ -0,0 +1,212 @@
+
+## Where are tests run?
+
+### CircleCI
+
+CircleCI is a service that automatically runs FxA tests in parallel jobs for every Pull Request and merge to `master` branch on GitHub. Scripts and configuration for CircleCI mostly lives in the `.circleci` directory.
+
+The file `.circleci/config.yml` contains general configuration, job definitions, and workflows orchestrating all the jobs.
+
+#### Important scripts
+
+##### .circleci/modules-to-test.js
+
+Running all tests across all modules / packages in FxA can be time-comsuming. Time can be saved by skipping tests that we know are unrelated to changes in some given commit. Applying that logic, this script outputs a name of individual packages that should be tested - or `all` if it decides that everything needs to be run.
+
+For master branch and other non-PR commits, the output is always `all`.
+
+For pull requests, the script fetches and parses the diff for the current PR. The URL for the diff is constructed from CircleCI env vars.
+
+If there's an error while fetching or parsing the diff, `all` is output. Otherwise, the script checks the path of each file changed to build a list of modules for testing.
+
+Some modules are depend on other modules. Within `package.json` at the root of the FxA project, there is an `fxa.moduleDependencies` object. This maps named modules to their dependencies.
+
+If the diff contains changes to any of the dependencies, the dependent module is also added to the list. So, for example, `fxa-content-server` lists `fxa-auth-server` as a dependency - thus, a change to `fxa-auth-server` causes `fxa-content-server` to also be included in the list.
+
+```json
+"fxa": {
+  "moduleDependencies": {
+    "fxa-content-server": [
+      "fxa-auth-server"
+    ]
+  }
+}
+```
+
+Finally, the list is output to the stdout, one module per line for each in need of testing.
+
+##### .circleci/build-test-deploy.sh {TEST_SCRIPT_NAME}
+
+Runs `build.sh`, `test.sh`, and `deploy.sh` for a given package - deriving the package name from the current working directory. Passes along the first parameter as the second parameter in `test.sh`.
+
+##### .circleci/build.sh {MODULE}
+
+Common build script for most modules.
+
+Logic for running or skipping build for a given module is implemented here: `packages/test.list` is expected to contain a list output from `modules-to-test.js` - if the current module's name is missing and the list isn't `all`, this script will exit immediately.
+
+Parameters are just `$1` for the name of the module being tested.
+
+The file `packages/version.json` is expected to have been generated earlier - i.e. from the `install` job in CircleCI. This file is copied into the package.
+
+For modules where `Dockerfile` is present, a container tagged `{MODULE}:build` is built from `Dockerfile`.
+
+For modules where `Dockerfile-build` is present, the container is built from that file instead.
```

Comment:
Not for this PR, but why are there both Dockerfile and Dockerfile-build files?

Reply:
Good question! 🤷‍♂

Reply:
No good reason. Just an quirk of the previous multi-repos